testing/trino-product-tests/src/main/java/io/trino/tests/product/hudi/TestHudiSparkCompatibility.java

```diff
@@ -288,6 +289,40 @@ public void testCopyOnWriteTableSelectWithSessionProperties()
         }
     }
 
+    @Test(groups = {HUDI, PROFILE_SPECIFIC_TESTS})
+    public void testTimelineTable()
+    {
+        String tableName = "test_hudi_timeline_system_table_" + randomNameSuffix();
+        String nonExistingTableName = tableName + "_non_existing";
+        createNonPartitionedTable(tableName, COW_TABLE_TYPE);
+        try {
+            assertThat(onTrino().executeQuery(format("SELECT action, state FROM hudi.default.\"%s$timeline\"", tableName)))
+                    .containsOnly(row("commit", "COMPLETED"));
+            assertQueryFailure(() -> onTrino().executeQuery(format("SELECT * FROM hudi.default.\"%s$timeline\"", nonExistingTableName)))
+                    .hasMessageMatching(".*Table 'hudi.default.test_hudi_timeline_system_table_.*_non_existing\\$timeline' does not exist");
```

Comment:
This is already covered in `TestHudiSystemTables`. Let's remove it and focus on the positive case.